openapi/generator.go

```diff
@@ -1276,6 +1276,13 @@ func parseExampleValue(t reflect.Type, value string) (interface{}, error) {
 		return strconv.ParseFloat(value, t.Bits())
 	case reflect.Ptr:
 		return parseExampleValue(t.Elem(), value)
+	case reflect.Struct:
```

Comment:
> Wondering again, since we now provide an interface, we should not restrict it to Struct kind. Instead, we should honor the ParseExemple method of every type that implement the Exampler and call it to get the example value.

Fixed.

> For types with a Struct kind, we would default to a call to the ParseExample method, if its implemented, otherwise return an empty string to signal that there is no example.

Hmm, I'm not a fan of an empty string. I think it's better if we return the error (like we currently do). If someone provided an `example` tag on a struct field, and the struct does not implement `Exampler` I think an error is more appropriate than defaulting to empty string. WDYT?

Reply:
Alright. The error doen't stop the spec generation anyway, so it's fine.